src/toast/templates/hwpss.py

```diff
@@ -0,0 +1,534 @@
+                    np.multiply(times[good], sincos[good, 2 * hc + 0]),
+                )
+                cov[4 * hr + 0, 4 * hc + 2] = np.dot(
+                    sincos[good, 2 * hr + 0], sincos[good, 2 * hc + 1]
+                )
+                cov[4 * hr + 0, 4 * hc + 3] = np.dot(
+                    sincos[good, 2 * hr + 0],
+                    np.multiply(times[good], sincos[good, 2 * hc + 1]),
+                )
+
+                cov[4 * hr + 1, 4 * hc + 0] = np.dot(
+                    np.multiply(times[good], sincos[good, 2 * hr + 0]),
+                    sincos[good, 2 * hc + 0],
+                )
+                cov[4 * hr + 1, 4 * hc + 1] = np.dot(
+                    np.multiply(times[good], sincos[good, 2 * hr + 0]),
+                    np.multiply(times[good], sincos[good, 2 * hc + 0]),
+                )
+                cov[4 * hr + 1, 4 * hc + 2] = np.dot(
+                    np.multiply(times[good], sincos[good, 2 * hr + 0]),
+                    sincos[good, 2 * hc + 1],
+                )
+                cov[4 * hr + 1, 4 * hc + 3] = np.dot(
+                    np.multiply(times[good], sincos[good, 2 * hr + 0]),
+                    np.multiply(times[good], sincos[good, 2 * hc + 1]),
+                )
+
+                cov[4 * hr + 2, 4 * hc + 0] = np.dot(
+                    sincos[good, 2 * hr + 1], sincos[good, 2 * hc + 0]
+                )
+                cov[4 * hr + 2, 4 * hc + 1] = np.dot(
+                    sincos[good, 2 * hr + 1],
+                    np.multiply(times[good], sincos[good, 2 * hc + 0]),
+                )
+                cov[4 * hr + 2, 4 * hc + 2] = np.dot(
+                    sincos[good, 2 * hr + 1], sincos[good, 2 * hc + 1]
+                )
+                cov[4 * hr + 2, 4 * hc + 3] = np.dot(
+                    sincos[good, 2 * hr + 1],
+                    np.multiply(times[good], sincos[good, 2 * hc + 1]),
+                )
+
+                cov[4 * hr + 3, 4 * hc + 0] = np.dot(
+                    np.multiply(times[good], sincos[good, 2 * hr + 1]),
+                    sincos[good, 2 * hc + 0],
+                )
+                cov[4 * hr + 3, 4 * hc + 1] = np.dot(
+                    np.multiply(times[good], sincos[good, 2 * hr + 1]),
+                    np.multiply(times[good], sincos[good, 2 * hc + 0]),
+                )
+                cov[4 * hr + 3, 4 * hc + 2] = np.dot(
+                    np.multiply(times[good], sincos[good, 2 * hr + 1]),
+                    sincos[good, 2 * hc + 1],
+                )
+                cov[4 * hr + 3, 4 * hc + 3] = np.dot(
+                    np.multiply(times[good], sincos[good, 2 * hr + 1]),
+                    np.multiply(times[good], sincos[good, 2 * hc + 1]),
+                )
+
+        # Fill in lower triangle
+        for hr in range(0, 4 * n_harmonics):
+            for hc in range(0, hr):
+                cov[hr, hc] = cov[hc, hr]
+        # LU factorization for later solve
```

Comment:
Should we check the condition number for pathological systems?

Reply:
Done- for observations where this covariance is poorly conditioned, the template amplitude are flagged.